exchanges/graphcache/src/store/store.ts

```diff
@@ -125,13 +148,63 @@ export class Store<
     return key ? `${data.__typename}:${key}` : null;
   }
 
+  prepareCache(type: OperationType): null | (() => void) {
+    const op = getCurrentOperation();
+    if (!op) {
+      initDataState(type, this.data);
+      return () => {
+        if (type === 'write') {
+          const deps = getCurrentDependencies();
+          const pendingOperations: Set<number> = new Set();
+
+          if (deps) {
+            // Collect operations that will be updated due to cache changes
+            for (const dep in deps) {
+              const keys = this.deps[dep];
+              if (keys) {
+                this.deps[dep] = [];
+                for (let i = 0, l = keys.length; i < l; i++) {
+                  pendingOperations.add(keys[i]);
+                }
+              }
+            }
+          }
+
+          pendingOperations.forEach(key => {
+            const op = this.operations.get(key);
+            if (op) {
+              this.operations.delete(key);
+              let policy: RequestPolicy = 'cache-first';
+              if (this.requestedRefetch.has(key)) {
+                this.requestedRefetch.delete(key);
+                policy = 'cache-and-network';
+              }
+              this.client!.reexecuteOperation(toRequestPolicy(op, policy));
+            }
+          });
+        }
+        clearDataState();
+      };
+    }
+
+    return null;
+  }
```

Comment:
This is an interesting one! So, we basically now have a system whereby `store` holds a lot more power over the reexecution behaviour. My gut feeling is that it should be possible for us to get rid of `executePendingOperations` and `collectPendingOperations` entirely somehow, but I think I'll need a closer look at the whole files to see whether that's indeed the case 🤔

Reply:
I honestly would prefer that as well, this logic is quite shared currently with the main exchange code.